### PR TITLE
support remove_input_padding for BertForSequenceClassification models

### PR DESCRIPTION
**Pull request by Altair-Alpha**
Related issue: https://github.com/NVIDIA/TensorRT-LLM/issues/1755
Content:
1. Support remove_input_padding for BertForSequenceClassification models (implementation details given in code comment)
2. Refinement of the `build.py` script, e.g., the original script doen't have a input model dir parameter, and will init the model with random weights, which is not intuitive.
3. Since the model input is changed from 3 to 5, i.e. `input_ids, input_lengths, token_type_ids, position_ids, max_input_length`, I add a standalone `run_remove_input_padding.py` demo script, and show how to build them with only `input_ids` and `token_type_ids`.

I only implemented and tested this for `BertForSequenceClassification` but **not other BERT models** yet, please feel free to do further work on this :)

### CODE REVIEW

**Comment by Altair-Alpha** [comment]
For our data, there's about 35% latency and 28% throughput enhancement when batch_size is 16, and the diff of output logits between the pytorch model and trt with remove_input_padding is below 0.01

**Comment by nv-guomingz** [comment]
Thanks @Altair-Alpha, we'll merge your changes and upstream to github next week.

File: examples/bert/build.py [hunk 1]
```diff
@@ -101,17 +103,91 @@ def parse_arguments():
                             'RobertaForQuestionAnswering',
                             'RobertaForSequenceClassification',
                         ])
+    parser.add_argument('--model_dir', type=str, required=True)
```
Comment (nv-guomingz) [inline examples/bert/build.py]: I suggest we set the required to False here for backward compatibilty or we need to add 
`        state_dict = torch.load(os.path.join(args.model_dir, "pytorch_model.bin"))
        hf_bert.load_state_dict(state_dict, strict=False)
` for all model type, right?

Comment (Altair-Alpha) [reply examples/bert/build.py]: Yes that makes sense. Making `--model_dir` optional and defaults to previous behavior.

**Comment by nv-guomingz** [comment]
hi @Altair-Alpha , we've merged your PR into our internal  code base with several minimal changes for internal ci.
This MR will be avaiable on next Tensor Tuesday.
